pinot-perf/src/main/java/org/apache/pinot/perf/BenchmarkNativeVsLuceneTextIndex.java

```diff
@@ -0,0 +1,214 @@
+import org.apache.pinot.spi.config.table.TableConfig;
+import org.apache.pinot.spi.config.table.TableType;
+import org.apache.pinot.spi.data.FieldSpec;
+import org.apache.pinot.spi.data.Schema;
+import org.apache.pinot.spi.data.readers.GenericRow;
+import org.apache.pinot.spi.data.readers.RecordReader;
+import org.apache.pinot.spi.utils.builder.TableConfigBuilder;
+import org.openjdk.jmh.annotations.Benchmark;
+import org.openjdk.jmh.annotations.BenchmarkMode;
+import org.openjdk.jmh.annotations.CompilerControl;
+import org.openjdk.jmh.annotations.Fork;
+import org.openjdk.jmh.annotations.Level;
+import org.openjdk.jmh.annotations.Measurement;
+import org.openjdk.jmh.annotations.Mode;
+import org.openjdk.jmh.annotations.OutputTimeUnit;
+import org.openjdk.jmh.annotations.Param;
+import org.openjdk.jmh.annotations.Scope;
+import org.openjdk.jmh.annotations.Setup;
+import org.openjdk.jmh.annotations.State;
+import org.openjdk.jmh.annotations.TearDown;
+import org.openjdk.jmh.annotations.Warmup;
+import org.openjdk.jmh.infra.Blackhole;
+import org.openjdk.jmh.runner.Runner;
+import org.openjdk.jmh.runner.options.OptionsBuilder;
+
+
+@BenchmarkMode(Mode.Throughput)
+@Fork(1)
+@Warmup(iterations = 5, time = 1)
+@Measurement(iterations = 5, time = 1)
+@OutputTimeUnit(TimeUnit.MINUTES)
+@State(Scope.Benchmark)
+public class BenchmarkNativeVsLuceneTextIndex {
+  private static final File INDEX_DIR = new File(FileUtils.getTempDirectory(), "TextSearchQueriesTest");
+  private static final String TABLE_NAME = "MyTable";
+  private static final String SEGMENT_NAME_NATIVE = "testSegmentNative";
+  private static final String DOMAIN_NAMES_COL = "DOMAIN_NAMES_COL";
+  private static final String INT_COL = "INT_COL";
+
+  private IndexSegment _indexSegment;
+
+  final String _luceneQuery =
+      "SELECT SUM(INT_COL) FROM MyTable WHERE TEXT_MATCH(DOMAIN_NAMES_COL, 'www.domain1%')";
+  final String _nativeQuery =
+      "SELECT SUM(INT_COL) FROM MyTable WHERE DOMAIN_NAMES_COL CONTAINS 'www.domain1.*'";
+  @Param("1000000")
+  int _numRows;
+  @Param({"0", "1", "10", "100"})
+  int _numBlocks;
+  @Param({"NATIVE", "LUCENE"})
+  private FSTType _fstType;
+
+  private PlanMaker _planMaker;
+  private QueryContext _queryContext;
+
+  @Setup(Level.Trial)
+  public void setUp()
+      throws Exception {
+    _planMaker = new InstancePlanMakerImplV2();
+    if (_fstType == FSTType.LUCENE) {
+      _queryContext = QueryContextConverterUtils.getQueryContextFromSQL(_luceneQuery);
+    } else {
+      _queryContext = QueryContextConverterUtils.getQueryContextFromSQL(_nativeQuery);
+    }
+    FileUtils.deleteQuietly(INDEX_DIR);
+
+    buildSegment(_fstType);
+
+    _indexSegment = loadSegment(_fstType);
+  }
+
+  @TearDown(Level.Trial)
+  public void tearDown() {
+    FileUtils.deleteQuietly(INDEX_DIR);
+  }
+
+  private List<String> getDomainNames() {
+    return Arrays.asList("www.domain1.com", "www.domain1.co.ab", "www.domain1.co.bc", "www.domain1.co.cd",
```

Comment:
Same here, we want to benchmark the text blocks instead of the single word case. Suggest replacing these domain names with texts